contracts/exchangeIssuance/ExchangeIssuance.sol

```diff
@@ -0,0 +1,244 @@
+pragma solidity >=0.6.10;
+pragma experimental ABIEncoderV2;
+
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+import "@uniswap/v2-periphery/contracts/interfaces/IUniswapV2Router02.sol";
+import "@uniswap/v2-core/contracts/interfaces/IUniswapV2Factory.sol";
+import "../lib/PreciseUnitMath.sol";
+import "../../external/contracts/UniswapV2Library.sol";
+import "../../external/contracts/SushiswapV2Library.sol";
+import "../interfaces/ISetToken.sol";
+import "../interfaces/IBasicIssuanceModule.sol";
+import "../interfaces/IWETH.sol";
+
+/**
+ * @title ExchangeIssuance
+ * @author Noah Citron
+ *
+ * Contract for minting and redeeming any Set token using
+ * ETH as the paying/receiving currency. All swaps are done using the best price
+ * found on Uniswap or Sushiswap.
+ *
+ */
+contract ExchangeIssuance {
+
+    using SafeMath for uint256;
+    
+    /* ============ State Variables ============ */
+
+    IUniswapV2Router02 private uniRouter;
+    address private uniFactory;
+    IUniswapV2Router02 private sushiRouter;
+    address private sushiFactory;
+
+    IBasicIssuanceModule private basicIssuanceModule;
+    address private WETH;
+
+    /* ============ Events ============ */
+
+    event ExchangeIssue(address indexed recipient, address indexed index, address indexed inputToken, uint256 amount);
+    event ExchangeRedeem(address indexed recipient, address indexed index, address indexed outputToken, uint256 amount);
+
+    /* ============ Constructor ============ */
+
+    constructor(address _uniFactory, address _uniRouter, address _sushiFactory, address _sushiRouter, address _basicIssuanceModule) public {
```

Comment:
I believe you could do the following `IUniswapV2Router02 _uniRouter` if you like this syntactically